pandas/core/indexes/base.py

```diff
@@ -2320,6 +2320,11 @@ def symmetric_difference(self, other, result_name=None):
         except TypeError:
             pass
 
+        # On equal MultiIndexes the difference is empty. Therefore an empty
+        # MultiIndex is returned GH13490
+        if self.nlevels > 1 and len(the_diff) == 0:
+            return type(self)([[] for _ in range(self.nlevels)],
```

Comment:
@jreback what do you think about a special case in `MulltiIndex._shallow_copy` when `values` is length 0 (that's what is passed from `symmetric_difference`)?

```python
    @Appender(_index_shared_docs['_shallow_copy'])
    def _shallow_copy(self, values=None, **kwargs):
        if values is not None:
            if 'name' in kwargs:
                kwargs['names'] = kwargs.pop('name', None)
            # discards freq
            kwargs.pop('freq', None)
            # this if block is new
            if len(values) == 0:
                return MultiIndex(levels=[[] for _ in range(self.nlevels)],
                                  labels=[[] for _ in range(self.nlabels)])
            return MultiIndex.from_tuples(values, **kwargs)
        return self.view()
```

this would "work" but I don't know if "array of length 0" means same structure, but empty. Maybe it's ok.